packages/web-player-component/src/index.js

```diff
@@ -36,31 +37,85 @@ export default class PlayerComponent extends HTMLElement {
       player: this.player,
       castAppId: {}
     });
+
+    // initiate EPAS analytics unless in incognito mode
+    this.playerAnalytics = null;
+    if (!this.hasAttribute('incognito')) {
+      const epasUrl = this.getAttribute('epas-url') || "https://sink.epas.eyevinn.technology";
+      this.playerAnalytics = new PlayerAnalyticsConnector(epasUrl);
+    }
+    
+    this.player.on(PlayerEvent.ERROR, ({ errorData, fatal }) => {
+      console.error('player reported error', errorData);
+      if (this.playerAnalytics) {
+        if (fatal) {
+          this.playerAnalytics.reportError(errorData);
+        } else {
+          this.playerAnalytics.reportWarning(errorData);
+        }
+      }
+      if (fatal) {
+        this.player.destroy();
+        console.log('player destroyed due to error');
+      }
+    });
+
+    this.player.on(PlayerEvent.BITRATE_CHANGE, (data) => {
+      if (this.playerAnalytics) {
+        this.playerAnalytics.reportBitrateChange({
+          bitrate: data.bitrate / 1000,
+          width: data.width,
+          height: data.height,
+        });
+      }
+    });
+  }
+
+  async analyticsInit() {
+    if (this.playerAnalytics) {
+      try {
+        await this.playerAnalytics.init({
+          sessionId: `${window.location.hostname}-${Date.now()}`
+        });
+      } catch (err) {
+        console.error(err);
+        this.playerAnalytics.deinit();
+        this.playerAnalytics = null;
+      }
+    }
+  }
+
+  async analyticsLoad() {
```

Comment:
Nothing asynchronous in this method right?

Reply:
That's correct, I'll fix